src/main/scala/com/fulcrumgenomics/util/Metric.scala

```diff
@@ -70,17 +70,17 @@ object Metric {
     * line should be a single metric. */
   def read[T <: Metric](lines: Iterator[String], source: Option[String] = None)(implicit tt: ru.TypeTag[T]): Seq[T] = {
     if (lines.isEmpty) throw new IllegalArgumentException(s"No header found in metrics" + source.map(" in source: " + _).getOrElse(""))
-    val names = lines.next().split(Delimiter)
-    if (lines.isEmpty) Seq.empty
-    else {
-      lines.zipWithIndex.map { case (line, lineNumber) =>
-        val values = line.split(Delimiter)
-        if (names.length != values.length) {
-          throw new IllegalArgumentException(s"Line ${lineNumber+1} did not have the same length (${values.length}) as the header (${names.length}")
-        }
-        build[T](names.zip(values))
-      }.toSeq
-    }
+    val parser = new DelimitedDataParser(lines=lines, delimiter=Delimiter, ignoreBlankLines=false, trimFields=true)
+    val names  = parser.headers
+    parser.zipWithIndex.map { case (row, lineNumber) =>
+      val values = names.indices.map { i => row[String](i) }.map { value =>
```

Comment:
I'm also confused that you're using the row.apply() here which doesn't return an option.  I think this is really a bug in DelimitedDataParser as I would expect that to throw an exception if there's no actual value for a field.  I think it would be better to use `row[String].get(i).getOrElse(ReflectionUtil.SpecialEmptyOrNoneToken)`